crates/bevy_ecs/src/component.rs

```diff
@@ -336,57 +408,68 @@ impl ComponentDescriptor {
         }
     }
 
-    fn new_non_send<T: Any>(storage_type: StorageType) -> Self {
+    /// Create a new `ComponentDescriptor` for a `!Send` resource.
+    ///
+    /// The [`StorageType`] for resources is always [`TableStorage`].
+    fn non_send_from_type<T: Any>() -> Self {
         Self {
             name: Cow::Borrowed(std::any::type_name::<T>()),
-            storage_type,
+            storage_type: StorageType::Table,
             is_send_and_sync: false,
             type_id: Some(TypeId::of::<T>()),
             layout: Layout::new::<T>(),
             drop: needs_drop::<T>().then_some(Self::drop_ptr::<T> as _),
         }
     }
-
-    #[inline]
-    pub fn storage_type(&self) -> StorageType {
-        self.storage_type
-    }
-
-    #[inline]
-    pub fn type_id(&self) -> Option<TypeId> {
-        self.type_id
-    }
-
-    #[inline]
-    pub fn name(&self) -> &str {
-        self.name.as_ref()
-    }
 }
 
-#[derive(Debug, Default)]
+/// The metadata store for all components within a [`World`].
+///
+/// A public read-only API is accessbile. To initialize a new component,
+/// use [`World::init_component`], [`World::init_resource`], [`World::init_non_send_resource`] or
+/// [`World::init_component_with_descriptor`].  Initialized components cannot be removed or
+/// altered from a world.
+///
+/// [`World`]: crate::world::World
+/// [`World::init_component`]: crate::world::World::init_component
+/// [`World::init_resource`]: crate::world::World::init_resource
+/// [`World::init_non_send_resource`]: crate::world::World::init_non_send_resource
+/// [`World::init_component_with_descriptor`]: crate::world::World::init_component_with_descriptor
+#[derive(Debug)]
 pub struct Components {
     components: Vec<ComponentInfo>,
-    indices: std::collections::HashMap<TypeId, usize, fxhash::FxBuildHasher>,
-    resource_indices: std::collections::HashMap<TypeId, usize, fxhash::FxBuildHasher>,
+    ids: HashMap<TypeId, ComponentId, fxhash::FxBuildHasher>,
+    resource_ids: HashMap<TypeId, ComponentId, fxhash::FxBuildHasher>,
```

Comment:
❓ Sorry if this is a super basic question, but I always assumed resources were something totally different from components. But here we are storing resource ids in the components struct, which makes me think they are the same thing. 

Are resources just components?

Reply:
We're lacking a good word for components + resources :) They share a lot of the same infrastructure, but the naming uses Component in a few places :(

Reply:
They still have the same IDs, ArchetypeComponentIds, etc. for controlling access and identification, even if the underlying methodology and storage is separated.